src/main/java/duke/command/DukeCommand.java

```diff
@@ -0,0 +1,248 @@
+                storage.saveTaskList(taskList);
+                ui.outputLine("I've removed the following task.");
+                ui.outputLine(task.toString());
+            }),
+    FIND_TASK("find",
+            "Find a task",
+            new DukeCommandConfig(new DukeCommandArgument("keyword", "The keyword to filter tasks by",
+                    DukeCommandArgumentType.REQUIRED), Map.of()),
+            (TaskList taskList, Ui ui, Storage storage, String arg, Map<String, String> namedArgs) -> {
+                List<DukeTask> dukeTasks = taskList.getTasks().stream()
+                        .filter(dukeTask -> dukeTask.name.toLowerCase().contains(arg.toLowerCase()))
+                        .collect(Collectors.toList());
+                TaskList filteredTaskList = new TaskList(dukeTasks);
+                ui.outputLine(String.format("Here are the tasks with titles containing \"%s\"", arg));
+                ui.printTaskList(filteredTaskList);
+            }),
+    EXIT("bye",
+            "Exit Duke",
+            DukeCommandConfig.NO_ARGUMENTS,
+            (TaskList taskList, Ui ui, Storage storage, String arg, Map<String, String> namedArgs) -> {
+                ui.markExit();
+            }),
+    MARK_DONE("done",
+            "Mark the task as done",
+            new DukeCommandConfig(new DukeCommandArgument("index", "The position of the task in the list",
+                    DukeCommandArgumentType.REQUIRED), Map.of()),
+            (TaskList taskList, Ui ui, Storage storage, String arg, Map<String, String> namedArgs) -> {
+                DukeTask task = taskList.getTaskAt(parseTaskIndex(taskList, arg));
+                if (task.isDone()) {
+                    ui.outputLine("The following task is already marked as done! Good job!");
+                } else {
+                    task.markAsDone();
+                    storage.saveTaskList(taskList);
+                    ui.outputLine("I've marked the following task as done!");
+                }
+                ui.outputLine(task.toString());
+            });
+
+    private final String command;
+    private final String description;
+    private final DukeCommandConfig config;
+    private final DukeCommandAction action;
+
+    DukeCommand(String command, String description, DukeCommandConfig config, DukeCommandAction action) {
+        this.command = command;
+        this.description = description;
+        this.config = config;
+        this.action = action;
+    }
+
+    public static Optional<DukeCommand> getClosestMatch(String command) {
+        return Arrays.stream(DukeCommand.values())
+                .sorted(Comparator.comparingInt(c -> -c.getName().length()))
+                .filter(c -> command.startsWith(c.getName()))
+                .findFirst();
+    }
+
+    /**
+     * Returns the name of the command.
+     * @return the name of the command
+     */
+    public String getName() {
```

Comment:
May I know what name is it referring to? The name of the command or do you mean the type of the command?

Reply:
nice to see that there has been javadoc comment added to explain it better.